server/service.go

```diff
@@ -279,6 +280,12 @@ func (m *BoostService) handleGetHeader(w http.ResponseWriter, req *http.Request)
 				return
 			}
 
+			// Verify response coherence with proposer's input data
+			if responsePayload.Data != nil && responsePayload.Data.Message.Header.ParentHash.String() != parentHashHex {
```

Comment:
`responsePayload.Data` nil check is already in line 261

```suggestion
			if responsePayload.Data.Message.Header.ParentHash.String() != parentHashHex {
```